src/game.cpp

```diff
@@ -7968,10 +7984,122 @@ bool pet_menu(monster *z)
 
         return true;
     }
+    if ( inject == choice ) {
+        auto filter = [] (const item &it) {
+            for (auto cont : it.contents) {
+                return cont.has_flag( "MUTAGEN_CEPHALOPOD" ) || cont.has_flag( "PURIFIER" );
```

Comment:
I assumed it already does that. May you show how me, how should I do it properly?